sql/2020/07_SEO/2019 Rewrites/core_web_vitals_by_device.sql

```diff
@@ -0,0 +1,75 @@
+ROUND(AVG( avg_lcp_adjusted) * 100, 2) AS pct_avg_lcp,
+ROUND(AVG( slow_lcp_adjusted) * 100, 2) AS pct_slow_lcp,
+
+ROUND(AVG( fast_fid_adjusted) * 100, 2) AS pct_fast_fid,
+ROUND(AVG( avg_fid_adjusted) * 100, 2) AS pct_avg_fid,
+ROUND(AVG( slow_fid_adjusted) * 100, 2) AS pct_slow_fid,
+
+ROUND(AVG( small_cls_adjusted) * 100, 2) AS pct_small_cls,
+ROUND(AVG( medium_cls_adjusted) * 100, 2) AS pct_medium_cls,
+ROUND(AVG( large_cls_adjusted) * 100, 2) AS pct_large_cls,
+
+ROUND(AVG( fast_ttfb_adjusted) * 100, 2) AS pct_fast_ttfb,
+ROUND(AVG( avg_ttfb_adjusted) * 100, 2) AS pct_avg_ttfb,
+ROUND(AVG( slow_ttfb_adjusted) * 100, 2) AS pct_slow_ttfb,
+
+ROUND(AVG( fast_fcp_adjusted) * 100, 2) AS pct_fast_fcp,
+ROUND(AVG( avg_fcp_adjusted) * 100, 2) AS pct_avg_fcp,
+ROUND(AVG( slow_fcp_adjusted) * 100, 2) AS pct_slow_fcp,
+
+# from 2019 07_05b - Older metrics for speed scores
+ROUND(COUNTIF(fast_fcp_adjusted >= .9 AND fast_fid_adjusted >= .95) * 100 / COUNT(0), 2) AS pct_fast,
+ROUND(COUNTIF(NOT(slow_fcp_adjusted >= .1 OR slow_fid_adjusted >= 0.05) AND NOT(fast_fcp_adjusted >= .9 AND fast_fid_adjusted >= .95)) * 100 / COUNT(0), 2) AS pct_avg,
+ROUND(COUNTIF(slow_fcp_adjusted >= .1 OR slow_fid_adjusted >= 0.05) * 100 / COUNT(0), 2) AS pct_slow
+
+FROM (
+    # fix percents so a rows fast+avg+slow total 1 
+    # needed when origins are split into multiple rows like device and country, all rows for an origin add up to 1. This changes it so each row adds up to 1
+    SELECT 
+        device, 
+
+        # re work out percentage for each row
+        SAFE_DIVIDE(fast_lcp, fast_lcp+avg_lcp+slow_lcp) AS fast_lcp_adjusted,
+        SAFE_DIVIDE(avg_lcp, fast_lcp+avg_lcp+slow_lcp) AS avg_lcp_adjusted,
+        SAFE_DIVIDE(slow_lcp, fast_lcp+avg_lcp+slow_lcp) AS slow_lcp_adjusted, 
+
+        SAFE_DIVIDE(fast_fid, fast_fid+avg_fid+slow_fid) AS fast_fid_adjusted,
+        SAFE_DIVIDE(avg_fid, fast_fid+avg_fid+slow_fid) AS avg_fid_adjusted,
+        SAFE_DIVIDE(slow_fid, fast_fid+avg_fid+slow_fid) AS slow_fid_adjusted, 
+
+        SAFE_DIVIDE(small_cls, small_cls+medium_cls+large_cls) AS small_cls_adjusted,
+        SAFE_DIVIDE(medium_cls, small_cls+medium_cls+large_cls) AS medium_cls_adjusted,
+        SAFE_DIVIDE(large_cls, small_cls+medium_cls+large_cls) AS large_cls_adjusted,
+
+        SAFE_DIVIDE(fast_ttfb, fast_ttfb+avg_ttfb+slow_ttfb) AS fast_ttfb_adjusted,
+        SAFE_DIVIDE(avg_ttfb, fast_ttfb+avg_ttfb+slow_ttfb) AS avg_ttfb_adjusted,
+        SAFE_DIVIDE(slow_ttfb, fast_ttfb+avg_ttfb+slow_ttfb) AS slow_ttfb_adjusted,
+
+        SAFE_DIVIDE(fast_fcp, fast_fcp+avg_fcp+slow_fcp) AS fast_fcp_adjusted,
+        SAFE_DIVIDE(avg_fcp, fast_fcp+avg_fcp+slow_fcp) AS avg_fcp_adjusted,
+        SAFE_DIVIDE(slow_fcp, fast_fcp+avg_fcp+slow_fcp) AS slow_fcp_adjusted
+
+    FROM `chrome-ux-report.materialized.device_summary` 
+    WHERE yyyymm = 202006 # can determine latest month via latest entry in chrome-ux-report
```

Comment:
Looks like August data is there now:

```suggestion
    WHERE yyyymm = 202008 # can determine latest month via latest entry in chrome-ux-report
```